datafusion/physical-expr-common/src/binary_view_map.rs

```diff
@@ -306,6 +323,76 @@ where
         }
     }
 
+    /// Retrieves the payloads for each value from `values`, either by using
+    /// small value optimizations or larger value handling.
+    ///
+    /// This function will compute hashes for each value and attempt to retrieve
+    /// the corresponding payload from the map. If the value is not found, it will return `None`.
+    ///
+    /// # Arguments:
+    ///
+    /// `values`: The array whose payloads need to be retrieved.
+    ///
+    /// # Returns
+    ///
+    /// A vector of payloads for each value, or `None` if the value is not found.
+    pub fn get_payloads(self, values: &ArrayRef) -> Vec<Option<V>> {
```

Comment:
maybe we could call this (and the other function) `into_values()`  as that is what it appears to be doing (consuming the map to get the values)

Reply:
I considered that as well, but when comparing `into_values` with `into_state` (which literally converts the map to a state), using the name `into_values` feels a bit off to me. Since it only returns the values’ state without changing the map.